### PR TITLE
rasusa 0.1.0 (new formula)

### PR DESCRIPTION
**Pull request by mbhall88**
- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

The only audit check it doesn't pass is `Stable: version 0.1.0 is redundant with version scanned from URL` as I have manually specified the version number.

Closes https://github.com/mbhall88/rasusa/issues/13

### CODE REVIEW

**Comment by mbhall88** [comment]
The checks seem to fail now purely based on the audit check mentioned above about the version number. If I remove the version specification it doesn't parse out the correct version, so it has to be specified.

File: Formula/rasusa.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    url "https://github.com/mbhall88/rasusa/releases/download/0.1.0/rasusa-0.1.0-x86_64-unknown-linux-musl.tar.gz"
+    sha256 "eb6d1c1ab48bd3fd568fd7dad338352c54a514762502fc2ec4172ae792a77832"
+  end
+
+  def install
+    bin.install "rasusa"
+  end
+
+  test do
+    system "#{bin}/rasusa", "--help"
```
Comment (sjackman) [inline Formula/rasusa.rb]:
```suggestion
    assert_match "Usage", shell_output("#{bin}/rasusa --help")
```

**Comment by sjackman** [comment]
> The checks seem to fail now purely based on the audit check mentioned above about the version number. If I remove the version specification it doesn't parse out the correct version, so it has to be specified.

That's quite strange. The version number `0.1.0` seems correct on CircleCI.
https://app.circleci.com/jobs/github/brewsci/homebrew-bio/7946

**Comment by sjackman** [comment]
I'd much prefer to build this program from source using the brewed `rust` as a build dependency.
```ruby
depends_on "rust" => :build
```

**Comment by mbhall88** [comment]
> I'd much prefer to build this program from source using the brewed `rust` as a build dependency.
> 
> ```ruby
> depends_on "rust" => :build
> ```

 My reasoning for doing it this way though was that
1. The user doesn't need to have `rust` installed this way
2. This is the same brew strategy used by [`ripgrep`](https://github.com/BurntSushi/ripgrep/blob/master/pkg/brew/ripgrep-bin.rb) which is a very heavily used rust tool.

I can change it to build from source with `cargo` if you like though.

**Comment by mbhall88** [comment]
@sjackman so this won't be merged unless I build from source?

**Comment by tseemann** [comment]
@mbhall88 I think @sjackman is just hoping we can start to migrate `rust` and `go` tools to be built from source now. it wasn't possible in the past as the compilers weren't properly within brew. the user doesn't need the compilers installed on their desktop, because the will still install binaries, but the CI/TRAVIS build system will compile from source rather than download Brent's binaries. 

I'm not against merging this as-is.  @sjackman ? @jonchang ?

**Comment by jonchang** [comment]
Think it should build from source rather than install from binaries. I’m with @sjackman on this one.

Bottles shouldn’t need rust as a run time dependency.

**Comment by mbhall88** [comment]
Ok, no worries. I have no idea how to make brew recipes (I effectively copied this from another rust tool) so I will find another one that builds from source and update this PR.

**Comment by mbhall88** [comment]
Right, I _think_ this recipe should do it. Please let me know if there is anything else I should be doing in the recipe.

**Comment by tseemann** [comment]
@mbhall88 don't worry, we all just copy other recipes - it's the only sane way to do it :)
And the build from source recipe is shorter, and will be easier to update version too.
Nice work!

**Comment by tseemann** [comment]
@mbhall88 thank you - it is now part of brew.

**Pull request closed without merging**